Modules/Microsoft365DSC/DSCResources/MSFT_TeamsGroupPolicyAssignment/MSFT_TeamsGroupPolicyAssignment.psm1

```diff
@@ -365,15 +365,15 @@ function Export-TargetResource
         $totalCount = $instances.Length
         foreach ($item in $instances)
         {
-            $Group = Find-CsGroup -SearchQuery $item.GroupId
+            [array]$Group = Find-CsGroup -SearchQuery $item.GroupId -ExactMatchOnly $true
             if ($null -eq $totalCount)
             {
                 $totalCount = 1
             }
             Write-Host "    |---[$j/$totalCount] GroupPolicyAssignment {$($Group.DisplayName)}" -NoNewline
             $results = @{
-                GroupDisplayName      = $Group.DisplayName
```

Comment:
Is there any chance, that the array Group does not hold any elements? If yes, we should add a null check and check for the lenght.

Reply:
I wondered if this was the case, but in this case it means that we haven't managed to find the resource with its ID.
And what to put in place ?
Or we throw an error in this case, like:
```
            [array]$Group = Find-CsGroup -SearchQuery $item.GroupId -ExactMatchOnly $true
            if ($null -eq $totalCount)
            {
                $totalCount = 1
            } else if ($totalCount -Eq 0)
            {
                throw "GPA {$($item.GroupId)}: Unable to find a CsGroup whis that ID"
            }

```
What do you prefer

Reply:
$totalCount is the number of instances and not related to the group count.

Throwing an exception will break the export. It would be better two write a warning/verbose message about an issue to resolve the group.

```powershell
if ($null -eq $group -or $group.length -eq 0)
{
# Write-Something
}
else if($group.lenght -ge 2){
#Write-something about several groups and taking the first one
}
else{
#current process
}
```

I haven't had a look into the return of Find-CsGroup and if there might be more than one return...

Reply:
If there is more than one, it's not an issue. They all have the same display name as we are using `ExactMatchOnly $true`, any of them is ok, so let's take the first.
For me the only non nominal case is `-eq 0`, that is not supposed to happen (we are retrieving the display name of a group by its UID). 
The only possibility I see is: if the group has been deleted concurrently between the retrieval of the list (L354) and the call to `Find-CsGroup` inside the loop L368.
I don't know what is the strategy for this kind of error handling in the DSC365 project.
Do you have a pointer to documentation about error handling or a piece of code having the same behaviour?

Reply:
I would be good to go with the easiest solution then and make sure to print out a warning to the log about the issue. This is something that should not fail the whole export.

Reply:
I'm not fully clear how do you 'print out' a warning following your standard.
I've pushed a new commit with a try.

Reply:
Are you expected something else ?

Reply:
Yes, this looks good. Just one minor thing: Could you Write the message to the verbose stream and not the host. Host might break in some unattended instances.

Reply:
Which one ?
I don't add any Write-Host on my code, I only changed the message of an existing one.
And I added a Write-Error